app/services/source_detection_service.py

```
@@ -0,0 +1,1191 @@
+    
+    def _extract_video_id(self, video_url: str) -> Optional[str]:
+        """Extract YouTube video ID from URL"""
+        try:
+            parsed_url = urlparse(video_url)
+            
+            if parsed_url.hostname in ['www.youtube.com', 'youtube.com']:
+                if parsed_url.path == '/watch':
+                    query_params = parse_qs(parsed_url.query)
+                    return query_params.get('v', [None])[0]
+                elif parsed_url.path.startswith('/embed/'):
+                    return parsed_url.path.split('/embed/')[1].split('?')[0]
```

Pylint (3.3.7)

[refactor] Unnecessary "elif" after "return", remove the leading "el" from "elif"

(R1705)